app/Repository/Eloquent/AuthRepository.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Repository\Eloquent;
+
+use App\Models\User;
+use App\Repository\BaseRepository;
+
+class AuthRepository extends BaseRepository{
+
+    protected $user;
+
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    public function signUp($request){
+        $request['password'] = bcrypt($request['password']);
+
+        if(!$user = $this->user->create($request)){
+            return [
+                "status" => $this->isUnsuccessful(),
```

Comment:
All this shouldn't happen in your repository - remember last time I said this approach wasn't ideal but I allowed it anyways - your repository should create a model and return that model, that's all.

Reply:
```
public function signUp($data){
    $data['password'] = bcrypt($data['password']);
   $this->user->create($data);
   
   //log in the user here
}
```